### ISSUE TITLE
Decoder for V2

### ISSUE THREAD

**Issue by SleimanJneidi** [InteractiveAdvertisingBureau/iabtcf-java#6]
I can see that there is not a decoder for consent strings in here, I am happy to work on it if you don't mind

**Comment by chrispaterson** [comment]
Do you mean encoder?

**Comment by SleimanJneidi** [comment]
No a decoder, a one that parses binary blob to a VendorConsent object

**Comment by chrispaterson** [comment]
@SleimanJneidi the only functionality currently is decoding a tc string, not encoding.

**Comment by SleimanJneidi** [comment]
@chrispaterson thanks for your response but where is that happening in the code?

**Comment by PauloMiranda98** [comment]
@SleimanJneidi to decode a TC String you will use:
`TCModel tcModel = TCString.getInstance().decode(tcStringBase64);`

As the TCModel object you can access any TC String attribute. To test you can generate TC String on this site: https://www.iabtcf.com/#/encode

Just remembering that this project is for version 2 of TC String.

**Comment by chrispaterson** [comment]
@SleimanJneidi @PauloMiranda98 we could really use some help with unit testing this library.  When I wrote the TypeScript library it was immensely helpful to prove that it actually works.  Could one of you set up the JUnit framework with stub tests?  I found it easiest to start with the field encoders https://github.com/InteractiveAdvertisingBureau/iabtcf-java/tree/master/src/main/java/com/iabtcf/encoder/field because they are small and discrete and then work up from there to larger logical constructs.
